jerry-core/jcontext/jcontext.h

```diff
@@ -187,6 +190,36 @@ extern jerry_hash_table_t jerry_global_hash_table;
 
 #endif /* !CONFIG_ECMA_LCACHE_DISABLE */
 
+#else /* JERRY_ENABLE_EXTERNAL_CONTEXT */
+
+#define JERRY_CONTEXT(field) (((jerry_context_t *) jerry_port_get_current_instance()->context_p)->field)
+
+#ifndef JERRY_SYSTEM_ALLOCATOR
+
+#define JERRY_HEAP_CONTEXT(field) (((jmem_heap_t *) jerry_port_get_current_instance ()->lcache_p)->field)
+
+#undef JMEM_HEAP_SIZE
+#define JMEM_HEAP_SIZE (jerry_port_get_current_instance ()->heap_size)
+
+#undef JMEM_HEAP_AREA_SIZE
+#define JMEM_HEAP_AREA_SIZE (jerry_port_get_current_instance ()->heap_size - JMEM_ALIGNMENT)
+
```

Comment:
On second thoughts, I don't think "one big slab of mem" is better than current jerry_instance_t (with 3 buffers)

1. > I guess another advantage of the "one big slab of mem" approach is that these getters would be part of jerry-core and the user would only have to provide a getter for that one big buffer.

Current jerry_instance_y can also support this. User just provide the getter for the instance, and other getters are in jerry-core.

2. > I wonder if it's desirable to "expose" that internally there is a separate struct for the context, the heap and the lcache.

Even if we use the one big buffer, we still needs the information of "heap size", "lcache size" or "offset of these 3 parts"  in the jerry_instance_p so that the helper function can find the context, heap, lcache. in other words, we have to expose the exists of these 3 structs.

3. current jerry_instance_t is more flexible, because the 3 buffers can from diff places, e.g. one is in heap, one is mapped from file, one is global array, ...